### PR TITLE
Multiple router conditions

### PR DESCRIPTION
**Pull request by DaigaroCota**
This pull request addresses vulnerabilities C2, C-3 and C-4 of Macro audit report.

### CODE REVIEW

**Comment by DaigaroCota** [comment]
This PR is waiting for #446.

**Review by pedrovalido — APPROVED** [review]
Very well done :) Ready to merge unless @brozorec 's answer regarding the scripts says otherwise

File: packages/protocol/script/02_SignBorrow.s.sol
Comment (pedrovalido) [inline packages/protocol/script/02_SignBorrow.s.sol:1]: Some tests need fixing
Screenshot 2023-05-03 at 15 24 01

**Review by pedrovalido — CHANGES_REQUESTED** [review]

File: packages/protocol/script/02_SignBorrow.s.sol
Comment (pedrovalido) [reply packages/protocol/script/02_SignBorrow.s.sol:1]: tests regarding the simple router have been fixed

[the remark this answers is not in the document]
Comment (DaigaroCota) [reply packages/protocol/script/02_SignBorrow.s.sol:1]: I updated the scrips to follow the new signing procedure that includes the `actionArgsHash`

**Comment by DaigaroCota** [comment]
@pedrovalido this is ready to be reviewed again.

**Pull request merged**